lib/net/ssh/proxy/command.rb

```diff
@@ -93,7 +93,7 @@ def io.recv(size)
             result = read_nonblock(size)
           rescue IO::WaitReadable, Errno::EINTR
             timeout_in_seconds = 20
-            if IO.select([self], nil, [self], timeout_in_seconds) == nil
+            if wait_readable(timeout_in_seconds) == nil
```

Comment:
Again looking at wait_readable it will not return nil. And it's defintely not equivalent to IO.select returning nil. IO.select returns nil on timeout, wait_readable returns nil on EOF according o 2.0 docs but 2.3 docs no longer says this explicitly: http://ruby-doc.org/stdlib-2.3.0/libdoc/io/wait/rdoc/IO.html#method-i-wait_readable.

http://ruby-doc.org/stdlib-2.0.0/libdoc/io/wait/rdoc/IO.html#method-i-wait_readable